native/python/jp_pythontypes.cpp

```diff
@@ -458,3 +458,37 @@ JPPyBuffer::~JPPyBuffer()
 	if (m_Valid)
 		PyBuffer_Release(&m_View);
 }
+
+char *JPPyBuffer::getBufferPtr(std::vector<Py_ssize_t>& indices)
+{
+	char *pointer = (char*) m_View.buf;
+	// No shape is just a 1D array
+	if (m_View.shape == NULL)
+	{
+		return pointer;
+	}
+
+	// No strides is C contiguous ND array
+	if (m_View.strides == NULL)
```

Comment:
Out of curiosity, what if the user passes a Fortran aligned buffer?

Reply:
We request types in order based on what we can handle.  Strided, offset, c contiguous.  Assuming that the fortran buffer implements offset or strided we handle it properly.  If they dont implement either of those easy to do things than we would give a type error.

Numpy is a bit more thorough in probing.  But given i cant test it currently i didnt want to implement it.  I only implemented strided because python had the access pattern in docs.  

I debated a bit here as to whether to give it fancy things like transpose or reshape on transfer.  But decided that was the producers job.

Reply:
I dont think we should try to implement whole of NumPys feature set.

Reply:
Okay I will leave it as is for now.   If we come up with a better way to construct tests perhaps more probing will be needed.  I think I hit 97% of the use cases for working with libraries like numpy.